File: app/src/main/java/nic/goi/aarogyasetu/models/Converters.java
package nic.goi.aarogyasetu.models;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;

/**
 * @author Niharika.Arora
 */
public class Converters {
    @TypeConverter
    public static EncryptedInfo fromString(String value) {
        Type listType = new TypeToken<EncryptedInfo>() {
        }.getType();
        return new Gson().fromJson(value, listType);
    }

    @TypeConverter
    public static String fromArrayList(EncryptedInfo list) {
        Gson gson = new Gson();
        return gson.toJson(list);
    }
}

File: app/src/main/java/nic/goi/aarogyasetu/models/Converters.kt
package nic.goi.aarogyasetu.models

import androidx.room.TypeConverter

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

import java.lang.reflect.Type

/**
 * @author Niharika.Arora
 */
object Converters {
    @TypeConverter
    fun fromString(value: String): EncryptedInfo {
        val listType = object : TypeToken<EncryptedInfo>() {

        }.getType()
        return Gson().fromJson(value, listType)
    }

    @TypeConverter
    fun fromArrayList(list: EncryptedInfo): String {
        val gson = Gson()
        return gson.toJson(list)
    }
}